podman.spec.rpkg

```diff
@@ -114,6 +114,15 @@ Requires: gnupg
 
 This package contains system tests for %{name}
 
+%package tests-e2e
+Summary: End-to-End tests for %{name}
+Requires: %{name} = %{epoch}:%{version}-%{release}
```

Comment:
This would need to require golang as runtime dep if you need to run the tests.

Alternatively what you can do instead is to compile the e2e test into a single binary. `go test -c -o ./bin/e2etest ./test/e2e`, however this still requires a lot of test config file under `test/` to be present on the system and in the right location so likely very hard to manage.